homeassistant/components/husqvarna_automower/entity.py

```diff
@@ -120,3 +153,24 @@ class AutomowerControlEntity(AutomowerAvailableEntity):
     def available(self) -> bool:
         """Return True if the device is available."""
         return super().available and _check_error_free(self.mower_attributes)
+
+
+class WorkAreaControlEntity(AutomowerControlEntity):
+    """Replies available when the mower is connected."""
+
+    def __init__(
+        self,
+        mower_id: str,
+        coordinator: AutomowerDataUpdateCoordinator,
+        work_area_id: int,
+    ) -> None:
+        """Initialize AutomowerEntity."""
+        super().__init__(mower_id, coordinator)
+        self.work_area_id = work_area_id
+
+    @property
+    def work_area_attributes(self) -> WorkArea:
+        """Get the mower attributes of the current mower."""
+        if TYPE_CHECKING:
+            assert self.mower_attributes.work_areas is not None
+        return self.mower_attributes.work_areas[self.work_area_id]
```

Comment:
Also wondering, should we check if `self.work_area_id` is in `self.mower_attributes.work_areas` in the available property? so if its not there we render the entity unavailable?

Reply:
Not sure, if I understood that right. Do you mean, a case were one of the work areas is deleted by the user in the mower and HA ist not restarted, yet?

Reply:
Yes, in that case, this would raise an error I think, so should we add a

```py
@property
def available(self) -> bool:
    return super().available and self.work_area_id in self.mower_attributes.work_areas
```

And maybe we can move the automatic work area creation and deletion to runtime, which is awesome (but can be daunting, so its fine to move to a followup)

Reply:
Do you have an good example for that?

Reply:
I use this logic in `withings` sensor platform (but it has added complexity).

In essence what is needed is:
1. Add listener to the coordinator in `async_setup_entry` of the relevant platforms
2. Keep a list (or a set for easy adding and subtracting of identifiers) of identifiers of (in this case) work areas you have added
3. In the listener you compare the new data with the old data, and if you find old ones you call the entity registry to delete the entities (`airgradient` has this in the mutable entities like `number`) and if you find new ones, you do a call to `async_add_entities`

`tedee` also is a nice example, but they have the logic added in the coordinator, which might not fit well